examples/dart/test/serialization_test.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:test/test.dart';
+import 'package:realm_dart/realm.dart';
+
+part 'serialization_test.realm.dart';
+
+@RealmModel()
+class _Pet {
+  late String type;
+  late int numberOfLegs;
+  late DateTime birthDate;
+
+  late double? price;
+}
+
+main() {
+  final config = Configuration.local([Pet.schema]);
+  final realm = Realm(config);
+
+  // create pet spider object
+  final spider = Pet('Jumping Spider', 8, DateTime.utc(2024, 4, 10));
+
+  realm.write(() {
+    realm.add(spider);
+  });
+
+  // :snippet-start: serialize
+  // Serialize to ejson by passing the object as a paramter to the method or ...
+  EJsonValue serializeByParam = toEJson(spider);
+
+  // ... by calling the method directly on the object
+  EJsonValue serializeWithCall = spider.toEJson();
+  // :snippet-end:
+
+  print(serializeWithCall);
+
+  test('serialize', () {
+    print(serializeByParam);
+
+    EJsonValue different = spider.toEJson();
+    print('new');
+    print(different);
+
+    final birthDate = DateTime.utc(2024, 4, 10);
+
+    // make sure serialized value matches expected
+    expect(serializeByParam, {
+      'type': 'Jumping Spider',
+      'numberOfLegs': {'\$numberInt': '8'},
+      'birthDate': {
+        '\$date': {'\$numberLong': birthDate.millisecondsSinceEpoch.toString()}
+      },
+      'price': null,
+    });
+  });
+
+  test('deserialize', () {
+    // :snippet-start: deserialize
+    // Deserialize from ejson
+    Pet deserializeFromEjson = fromEJson<Pet>(serializeByParam);
```

Comment:
```suggestion
    final deserializeFromEjson = fromEJson<Pet>(serializeByParam);
```
or
```suggestion
    Pet deserializeFromEjson = fromEJson(serializeByParam);
```
The last is shorter, but perhaps a little less obvious. The point is that the type can be inferred.

BTW; something is off with the syntax highlighting.

Reply:
I went with the second for simpliity if there is no preference.